File: task-03/Java/subtask04.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class subtask04 {
    public static String createStarPattern(int n) {
        StringBuilder pattern = new StringBuilder();

        for (int i = 1; i <= n; i += 2) {
            String padding = " ".repeat((n - i) / 2);
            String stars = "*".repeat(i);
            pattern.append(padding).append(stars).append("\n");
        }

        for (int i = n - 2; i >= 1; i -= 2) {
            String padding = " ".repeat((n - i) / 2);
            String stars = "*".repeat(i);
            pattern.append(padding).append(stars).append("\n");
        }

        return pattern.toString();
    }

    public static void main(String[] args) throws IOException {
        String content = new String(Files.readAllBytes(Paths.get("/home/KID6/Desktop/AmFoss S2+/task-03/Java/input2.txt")));
        int n = Integer.parseInt(content.trim());

        String pattern = createStarPattern(n);
        Files.write(Paths.get("/home/KID6/Desktop/AmFoss S2+/task-03/Java/output2.txt"), pattern.getBytes());
    }
}

File: task-03/Javascript/subtask-04.js
const fs = require('fs');

function generateStarPattern(n) {
  let pattern = '';

  for (let i = 1; i <= n; i += 2) {
    let stars = '*'.repeat(i);
    let padding = ' '.repeat((n - i) / 2);
    pattern += padding + stars + '\n';
  }

  for (let i = n - 2; i >= 1; i -= 2) {
    let stars = '*'.repeat(i);
    let padding = ' '.repeat((n - i) / 2);
    pattern += padding + stars + '\n';
  }

  return pattern;
}

fs.readFile('input2.txt', 'utf8', (err, data) => {
  if (err) throw err;

  let n = parseInt(data.trim());
  let pattern = generateStarPattern(n);

  fs.writeFile('output2.txt', pattern, (err) => {
    if (err) throw err;
    console.log('The pattern has been saved to output2.txt');
  });
});
